File: ios/LangToggle/Shared/ChangeInterval.swift
import Foundation

enum ChangeInterval: String, CaseIterable, Identifiable {
    case onUnlock = "ON_UNLOCK"
    case everyHour = "EVERY_HOUR"
    case every6Hours = "EVERY_6_HOURS"
    case everyDay = "EVERY_DAY"
    case every3Days = "EVERY_3_DAYS"
    case everyWeek = "EVERY_WEEK"

    var id: String { rawValue }

    var label: String {
        switch self {
        case .onUnlock: return "Every unlock / wake"
        case .everyHour: return "Every hour"
        case .every6Hours: return "Every 6 hours"
        case .everyDay: return "Once a day"
        case .every3Days: return "Every 3 days"
        case .everyWeek: return "Once a week"
        }
    }

    var refreshMinutes: Int? {
        switch self {
        case .onUnlock: return nil
        case .everyHour: return 60
        case .every6Hours: return 360
        case .everyDay: return 1440
        case .every3Days: return 4320
        case .everyWeek: return 10080
        }
    }
}

File: lib/models/change_interval.dart
/// How often the language should be changed when the toggle is ON.
enum ChangeInterval {
  onUnlock,
  everyHour,
  every6Hours,
  everyDay,
  every3Days,
  everyWeek,
}

extension ChangeIntervalExtension on ChangeInterval {
  String get label {
    switch (this) {
      case ChangeInterval.onUnlock:
        return 'Every unlock / wake';
      case ChangeInterval.everyHour:
        return 'Every hour';
      case ChangeInterval.every6Hours:
        return 'Every 6 hours';
      case ChangeInterval.everyDay:
        return 'Once a day';
      case ChangeInterval.every3Days:
        return 'Every 3 days';
      case ChangeInterval.everyWeek:
        return 'Once a week';
    }
  }

  /// Duration in minutes, or null for "on unlock" (handled separately).
  int? get intervalMinutes {
    switch (this) {
      case ChangeInterval.onUnlock:
        return null;
      case ChangeInterval.everyHour:
        return 60;
      case ChangeInterval.every6Hours:
        return 360;
      case ChangeInterval.everyDay:
        return 1440;
      case ChangeInterval.every3Days:
        return 4320;
      case ChangeInterval.everyWeek:
        return 10080;
    }
  }

  String get key => name;

  static ChangeInterval fromKey(String key) {
    return ChangeInterval.values.firstWhere(
      (e) => e.key == key,
      orElse: () => ChangeInterval.everyDay,
    );
  }
}
